app/views/layouts/_nav.html.erb

```diff
@@ -0,0 +1,32 @@
+<nav class="navbar navbar-inverse">
+  <div class="container-fluid">
+    <div class="navbar-header">
+      <ul class="nav nav-pills">
+        <li role="presentation"><%= link_to "Home", root_path, style: "color:turquoise" %></li>
+        <li role="presentation"> <%= link_to "About", about_path, style: "color:turquoise" %></li>
+        <li role="presentation"> <%= link_to "Show All Posts", posts_path, style: "color:turquoise" %></li>
+        <li role="presentation"> <%= link_to "Create New Post", new_post_path, style: "color:turquoise" %></li>
+      </ul>
+    </div>
+    <div class="navbar-nav navbar-right">
+      <ul class="nav nav-pills">
+        <% if user_signed_in?  %>
+              <%# The link_to below will make <a> tag with data-display-favorite attribute.  %>
+              <%# This is good when using javascript to access the data %>
+              <%# link_to "My Favourite Posts", posts_path(), data: {display_favourite: true}   %>
+              <%# The link_to method below will create params[:display_favourite, and set it to true %>
+              <li><%= link_to "My Favourite Posts", posts_path(display_favourite: true)   %></li>
```

Comment:
Cool! So this will just filter your index page to just show posts that the current user has favourited right? I had a completely new page. Mind you I didn't have search implemented at all but still, this is a much better solution in my opinion.